### PR TITLE
Show version information

### PR DESCRIPTION
**Pull request by romanofski**
With this patch, users can invoke the program with --help to get version
information.

Part of: https://github.com/purebred-mua/purebred/issues/124

### CODE REVIEW

**Comment by romanofski** [comment]
I want to create a new Fedora build, but would like to add in the version information. This patch doesn't show version information in the UI, but only if you invoke --help. I first thought about also implementing a way to show it in the UI, but ignored it for now.

File: src/Purebred.hs [hunk 1]
```diff
@@ -145,7 +147,11 @@ appconfig =
 optParser :: ParserInfo AppConfig
 optParser = info
   (appconfig <**> helper)
-  (fullDesc <> progDesc "purebred" <> header "a search based, terminal mail user agent")
+  (fullDesc
+   <> progDesc "purebred"
+   <> header ("a search based, terminal mail user agent - " <> versionString))
+  where
+    versionString = showVersion version
```
Comment (frasertweedale) [inline src/Purebred.hs]: We might as well add a `--version` option while we're at it?

Comment (romanofski) [reply src/Purebred.hs]: Fair enough... will update

**Comment by romanofski** [comment]
Alright here we go. Was tempted to include more information from https://hackage.haskell.org/package/Cabal-2.2.0.1 but then shied away from depending on it.

**Comment by frasertweedale** [comment]
@romanofski what other information is available with Cabal-2.2?

Anyhow, ACK.  There are some tweaks I want to make bug will tackle later.

**Comment by romanofski** [comment]
Ooh.. `buildArch` was one I thought would be interesting ...

**Pull request merged**